mrmustard/physics/representations.py

```diff
@@ -774,6 +794,15 @@ def __matmul__(self, other: Fock) -> Fock:
                 batched_array.append(math.tensordot(reduced_s.array[i], reduced_o.array[j], axes))
         return self.from_ansatz(ArrayAnsatz(batched_array))
 
+    def serialize(self) -> list[tuple[str, ArrayLike]]:
+        """Serialize a Fock instance."""
+        return [("array", self.data)]
```

Comment:
I wonder if there is some way to set this up such that we have a generic serialize class that everything serializable inherits from and we define the attrs on each of these classes telling the serializer what to serialize (maybe through `__slots__`?). Something like that would streamline a lot of this I believe

Reply:
I didn't want to over-engineer this, so I felt that adhering to a simple standard was sufficient, the standard being that representations return a list that is ready to be turned into a dict. We could also do something like `__slots__`, but I'd prefer to leave room for some flexibility without making the code too ugly (and this feels quite compact to me 😄 )

I also considered having representations follow the same standard as circuit components (return two items: a serializable dict, and non-serializable data in this format), then have the circuit component merge its data with its representation's data. Decided against it because this is more compact and equally consistent. lmk if you have a strong preference any which way